apps/backend/src/datasources/stfc/StfcProposalDataSource.ts

```diff
@@ -19,10 +20,12 @@ import { ProposalsFilter } from './../../resolvers/queries/ProposalsQuery';
 import PostgresProposalDataSource from './../postgres/ProposalDataSource';
 import { StfcUserDataSource } from './StfcUserDataSource';
 
-const stfcUserDataSource = new StfcUserDataSource();
-
 @injectable()
 export default class StfcProposalDataSource extends PostgresProposalDataSource {
+  protected stfcUserDataSource: StfcUserDataSource = container.resolve(
+    Tokens.UserDataSource
+  ) as StfcUserDataSource;
```

Comment:
We can also inject it into the constructor.

Reply:
I could, but many of our data sources don't have an explicit constructor. If I were to implement one with the `@inject` annotation I'd have to also add annotations and parameters for the ones we'd then need to pass to the parent classes constructor, and it ends up being a fair bit more code than this.